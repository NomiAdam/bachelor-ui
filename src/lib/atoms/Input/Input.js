/* eslint-disable no-nested-ternary */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { path } from 'ramda';
import { noop } from 'ramda-extension';
import { greenValid, redError } from './constants/icon';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

export const StyledInput = styled.input`
	border: 0;
	font-family: inherit;
	font-size: 16px;
	font-weight: 500;
	width: 100%;
	border-bottom: 1px solid ${ resolveTheme(TYPES.COLOR) };
	${ ({ borderBackground, ...props }) => (borderBackground ? `border-color: ${ borderBackground };` : resolveTheme(TYPES.COLOR)(props)) }
	border-radius: 0;
	color: ${ resolveTheme(TYPES.COLOR) };
	transition: .15s ease all;
	background-image: ${ ({ displayIcon, error }) => (displayIcon ? error ? redError : greenValid : 'none') };
	background-position: right;
	background-repeat: no-repeat;
	background-origin: content-box;
	background-color: inherit;
	box-sizing: border-box;
	&:focus {
		outline: none;
	}
`;

const getTargetValue = path(['target', 'value']);
const Input = ({ placeholder, onChange, ...rest }) => (
    placeholder ? (
        <StyledInput
            { ...rest }
            onChange={ e => onChange(e, getTargetValue(e)) }
            placeholder={ placeholder }
        />
    ) : (
        <StyledInput { ...rest } onChange={ e => onChange(e, getTargetValue(e)) } placeholder="&nbsp;" />
    )
);

Input.propTypes = {
    /**
	 * Any value of Input
	 */
    value: PropTypes.any,
    /**
	 * Function triggered during onChange
	 */
    onChange: PropTypes.func,
    /**
	 * Whether Input is disabled or not
     */
    disabled: PropTypes.bool,
    /**
	 * Whether we have some kind of validation error
     */
    error: PropTypes.bool,
    /**
	 * Function to be called upon onBlur event on input field
     */
    onBlur: PropTypes.func,
    /**
	 * String representation of input type
     */
    type: PropTypes.string,
    /**
	 * Whether we should display icon in input or not
     */
    displayIcon: PropTypes.bool,
    /**
     * String indicating border-color css property
     */
    borderBackground: PropTypes.string,
    /**
     * String definition of placeholder text
     */
    placeholder: PropTypes.string,
    /**
   * String representation of name prop
   */
    name: PropTypes.string,
};

Input.defaultProps = {
    type: 'text',
    onChange: noop,
};

export default Input;
