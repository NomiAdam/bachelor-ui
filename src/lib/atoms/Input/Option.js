// eslint-disable-next-line no-unused-vars
import React from 'react';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const StyledOption = styled.option`
 	position: absolute;
	padding: 20px 2%;
    width: 96%;
    display: block;
    margin: 0px;
    cursor: pointer;
    background-color: transparent;
    font-size: 16px;
    transition: .5s all ease;
    &:hover {
    	background-color: ${ lightTheme.whiteBlue };
    };
`;

export default StyledOption;
