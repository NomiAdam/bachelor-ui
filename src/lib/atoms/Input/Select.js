// eslint-disable-next-line no-unused-vars
import React from 'react';
import styled from 'styled-components';
import { basicTheme } from '../../constants/theme';

export const StyledSelect = styled.select`
	height: 30px;
  	min-width: 50px;
  	padding: .25em;
  	border: 0 solid transparent;
  	border-radius: 3px;
  	border-bottom: ${ ({ withBorder }) => (withBorder ? `2px solid ${ basicTheme.blue }` : '0px') };
  	overflow-y: scroll;
  	font-weight: bold;
  	letter-spacing: .15em;
  	${ ({ error }) => (error ? `
	    border: 1px solid ${ basicTheme.red };
	    border-top-left-radius: 5px;
        border-top-right-radius: 5px;
	` : '') }
  	&:focus {
  		outline: none;
  	}
  	box-shadow: 1px 2px 10px -2px ${ basicTheme.grey };
`;

export default StyledSelect;
