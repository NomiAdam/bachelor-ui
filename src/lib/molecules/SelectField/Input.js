import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyledInput = styled.input`
    border: none;
    border-bottom: 1px solid ${ resolveTheme(TYPES.COLOR) };
    box-sizing: border-box;
    width: 100%;
	  min-height: 30px;
	  padding: 10px 20px;
	  font-size: 16px;
	  display: inline-block;
	  text-align: center;
	  background-color: ${ ({ disabled }) => (disabled ? lightTheme.lightGrey : 'transparent') };
	  &:hover {
	    cursor: ${ ({ disabled }) => (disabled ? 'initial' : 'pointer') };
	  }
    &:focus {
        outline: none;
    }
`;

const Input = ({
    value, handleFocus, handleChange, placeholder, disabled,
}) => (
    <StyledInput disabled={ disabled } onChange={ handleChange } onFocus={ handleFocus } value={ value } placeholder={ placeholder } />
);

Input.propTypes = {
    /**
     * Current value of Input field
     */
    value: PropTypes.any,
    /**
     * Function triggered when Input is Focused, ie. user clicked into Input
     */
    handleFocus: PropTypes.func,
    /**
     * Function to handle value change on user input
     */
    handleChange: PropTypes.func,
    /**
     * String placeholder value, for when Input is empty
     */
    placeholder: PropTypes.string,
    /**
     * Whether select field is disabled or not
     */
    disabled: PropTypes.bool,
};

export { StyledInput };
export default Input;
