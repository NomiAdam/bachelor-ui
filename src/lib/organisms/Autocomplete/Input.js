import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const StyledInput = styled.input`\n    border: none;\n    border-bottom: 1px solid ${ lightTheme.lightGrey };\n    box-sizing: border-box;\n    width: 100%;\n	min-height: 30px;\n	padding: 10px 20px;\n	font-size: 16px;\n    &:focus {\n        outline: none;\n    }\n`;

const Input = ({
    value, handleFocus, handleChange, placeholder, handleEnter,
}) => (
    <StyledInput onKeyPress={ handleEnter } onChange={ handleChange } onFocus={ handleFocus } value={ value } placeholder={ placeholder } />
);

Input.propTypes = {
    /**
     * Current value of Input field
     */
    value: PropTypes.string,
    /**
     * Function triggered when Input is Focused, ie. user clicked into Input
     */
    handleFocus: PropTypes.func,
    /**
     * Function to handle value change on user input
     */
    handleChange: PropTypes.func,
    /**
     * String placeholder value, for when Input is empty
     */
    placeholder: PropTypes.string,
    /**
     * Function to be called upon enter press
     */
    handleEnter: PropTypes.func,
};

export { StyledInput };
export default Input;
