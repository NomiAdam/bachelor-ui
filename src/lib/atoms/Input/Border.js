/* eslint-disable no-nested-ternary */
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { StyledInput } from './Input';
import { StyledTextArea } from './TextAreaInput';
import { StyledSelect } from './Select';
import { basicTheme, lightTheme } from '../../constants/theme';
import { maxMobileM } from '../../constants/device';

const StyledBorder = styled.span`
	position: absolute;
	bottom: ${ ({ select, textArea }) => (select ? '-8px' : textArea ? '4px' : '-6px') }
	left: 0;
	height: 2px;
	width: 100%;
	background: ${ ({ background, error }) => (error ? basicTheme.red : background || lightTheme.lightBlue) };
	transform: scaleX(0);
	transform-origin: 0 0;
	transition: .5s ease all;
	${ StyledInput }:focus ~ & {
		transform: scaleX(1);
	}
	${ StyledTextArea }:focus ~ & {
		transform: scaleX(1);
	}
	${ StyledSelect }:focus ~ & {
		transform: scaleX(1);
	}
	@media ${ maxMobileM } {
	  	bottom: ${ ({ select, textArea }) => (select ? '-8px' : textArea ? '4px' : '-4px') }
	}
`;

StyledBorder.propTypes = {
    /**
	 * String representing color of border, ie. "red", "blue", ....
	 */
    background: PropTypes.string,
    /**
	 * Whether we use border for Input
	 */
    select: PropTypes.bool,
    /**
	 * Whether we use border for TextArea
	 */
    textArea: PropTypes.bool,
    /**
     * Whether we have validation or other error with TextField
     */
    error: PropTypes.bool,
};

export default StyledBorder;
