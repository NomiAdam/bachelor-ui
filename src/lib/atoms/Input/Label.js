import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { StyledInput } from './Input';
import { StyledTextArea } from './TextAreaInput';
import { maxMobileM, maxMobileS } from '../../constants/device';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

export const StyledLabel = styled.span`
	position: absolute;
	top: ${ ({ multiline }) => (multiline ? '-7px' : '0px') };
	left: 0px;
	font-size: 16px;
	color: ${ resolveTheme(TYPES.COLOR) };
	font-weight: 500;
	transform-origin: 0 0;
	transition: 0.3s ease all;
	min-width: 200px;
	max-width: 200px;
	${ StyledInput }:focus ~ & {
		transform: translateY(${ ({ multiline }) => (multiline ? '-25px' : '-15px') }) scale(.75);
	}
	${ StyledInput }:not(:placeholder-shown) ~ & {
		transform: translateY(${ ({ multiline }) => (multiline ? '-25px' : '-15px') }) scale(.75);
	}
	${ StyledTextArea }:focus ~ & {
		transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-75px') }) scale(.75);
	}
	${ StyledTextArea }:not(:placeholder-shown) ~ & {
		transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-75px') }) scale(.75);
	}
	@media ${ maxMobileM } {
		top: ${ ({ multiline }) => (multiline ? '0px' : '5px') };
		font-size: 12px;
		${ StyledInput }:focus ~ & {
		transform: translateY(${ ({ multiline }) => (multiline ? '-21px' : '-18px') }) scale(.75);
		}
		${ StyledInput }:not(:placeholder-shown) ~ & {
			transform: translateY(${ ({ multiline }) => (multiline ? '-21px' : '-18px') }) scale(.75);
		}
		${ StyledTextArea }:focus ~ & {
			transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-77px') }) scale(.75);
		}
		${ StyledTextArea }:not(:placeholder-shown) ~ & {
			transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-77px') }) scale(.75);
		}
	}
	@media ${ maxMobileS } {
		top: ${ ({ multiline }) => (multiline ? '2px' : '8px') };
		font-size: 10px;
		${ StyledInput }:focus ~ & {
			transform: translateY(${ ({ multiline }) => (multiline ? '-20px' : '-18px') }) scale(.75);
		}
		${ StyledInput }:not(:placeholder-shown) ~ & {
			transform: translateY(${ ({ multiline }) => (multiline ? '-20px' : '-18px') }) scale(.75);
		}
		${ StyledTextArea }:focus ~ & {
			transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-79px') }) scale(.75);
		}
		${ StyledTextArea }:not(:placeholder-shown) ~ & {
		transform: translateY(${ ({ multiline, rows }) => (multiline ? `-${ rows * 18 }px` : '-79px') }) scale(.75);
		}
	}
`;

const Label = ({ label, multiline, rows }) => (
    <StyledLabel
        rows={ rows }
        multiline={ multiline }
    >
        {label}
    </StyledLabel>
);

Label.propTypes = {
    /**
     * String representing value of Label
     */
    label: PropTypes.string,
    /**
	 * Whether we are in multiline textField or not
     */
    multiline: PropTypes.bool,
    /**
   * Number of rows
   */
    rows: PropTypes.number,
};

Label.defaultProps = {
    label: '',
    rows: 10,
};

export default Label;
