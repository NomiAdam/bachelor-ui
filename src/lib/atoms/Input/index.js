import Input from './Input';
import Label from './Label';
import Border from './Border';
import TextAreaInput from './TextAreaInput';
import Select from './Select';
import Option from './Option';

export {
    Label,
    Border,
    TextAreaInput,
    Select,
    Option,
};

export default Input;
