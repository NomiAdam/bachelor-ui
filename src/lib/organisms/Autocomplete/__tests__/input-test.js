/* eslint-disable no-undef */
import React from 'react';
import Input, {StyledInput} from '../Input';
import {shallow} from 'enzyme';

const mockChange = jest.fn();
const mockFocus = jest.fn();

describe('AutocompleteInput', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    it('should be defined', () => {
        expect(Input).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Input/>)).toMatchSnapshot();
    });
    it('should match the snapshot with value', () => {
        expect(shallow(<Input value={'mock'} />)).toMatchSnapshot();
    });
    it('should be of type StyledInput', () => {
        expect(shallow(<Input/>).type()).toBe(StyledInput);
    });
    it('should not fire handleChange action on render', () => {
        shallow(<Input handleChange={mockChange}/>);
        expect(mockChange).not.toHaveBeenCalled();
    });
    it('should not fire handleFocus action on render', () => {
        shallow(<Input handleFocus={mockFocus}/>);
        expect(mockFocus).not.toHaveBeenCalled();
    });
    it('should fire handleFocus action on focus', () => {
        const wrapper = shallow(<Input handleFocus={mockFocus}/>);
        wrapper.simulate('focus');
        expect(mockFocus).toHaveBeenCalled();
    });
    it('should fire handleChange action on change', () => {
        const wrapper = shallow(<Input handleChange={mockChange}/>);
        wrapper.simulate('change');
        expect(mockChange).toHaveBeenCalled();
    });
});
