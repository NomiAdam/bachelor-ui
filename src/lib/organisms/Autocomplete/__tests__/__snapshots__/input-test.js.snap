// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteInput should match the snapshot 1`] = `<Input__StyledInput />`;

exports[`AutocompleteInput should match the snapshot with value 1`] = `
<Input__StyledInput
  value="mock"
/>
`;
